traffective.module

```diff
@@ -235,10 +248,9 @@ function traffective_node_view_alter(&$build) {
           $previous_offset = $paragraph_count;
           // Inject another ad slot if the previous was rendered to adjust the
           // amount to the article length.
-          $ad_slots['rectangle'][] = new AdSlot('rectangle', 5);
-          // Only place one ad to polizei articles.
-          if ($channel === 'Polizei') {
-            break 2;
+          if (!$is_polizei) {
+            // Only place one ad to polizei articles.
+            $ad_slots['rectangle'][] = new AdSlot('rectangle', 5);
```

Comment:
The previous code had a `break 2` here, causing the current loop as well as the outer paragraphs loop to be discontinued, effectively skipping over to the end of the function.

Not breaking here causes  the code to be re-executed for each dom node in the current paragraph as well as all following paragraphs.  It also causes the following code for parallax slots to be executed.

Is this change of behavior intended?

Reply:
From the Asana ticket:

> The rectangle should be placed after the second paragraph only once. Not again after the fourth or sixth.

Appears to confirm my concern?

Reply:
Absolutely, you are right:
 - The add should be placed only once after the 2nd <p> and that's all.
 - Any yes, now the whole loop is executed, because the both loops are building the `$build` which is the final render, so we can not skip it, we just have to prevent further ads.

Reply:
Good, so you're adding back the control structure logic using `break 2`?

Reply:
no, with break skips the loop and don't place further paragraphs in the rendered article. Whit the condition we only place one ad but all further paragraphs are added in the `$build` so the content of the article is outputed. The actual implementation is the right one for it.

Reply:
Okay, if you say so. 👍